src/task2/user-storage.ts

```diff
@@ -0,0 +1,61 @@
+import { UserList, User, UserSchema } from './types';
+import { mockUsers } from './mocks';
+
+export class UserStorage {
+  private users: UserList = {};
+
+  constructor(initialUsers: UserList) {
+    this.users = initialUsers;
+  }
+
+  isUserExists(id: User['id']): boolean {
```

Comment:
Since we're using uuid I don't think we would ever run into `User exists` issue. In your sample data I would also generate a couple of uuids to be consistent (instead of `some-id-2` etc..)

Reply:
by the way I think we should not allow users to modify id in any way. If they pass an id I would either throw an error or maybe just ignore that (with some optional warning returned)..

Reply:
I am already not allow user to do that. I have separate validator for updating user, that allows to pass only username, password and age. Any other fields will throw an error.